src/core/server/http/lifecycle/on_pre_response.ts

```diff
@@ -106,26 +121,30 @@ export function adoptToHapiOnPreResponseFormat(fn: OnPreResponseHandler, log: Lo
           : response.statusCode;
 
         const result = await fn(KibanaRequest.from(request), { statusCode }, toolkit);
-        if (!preResponseResult.isNext(result)) {
+
+        if (preResponseResult.isNext(result)) {
+          if (result.headers) {
+            if (isBoom(response)) {
+              findHeadersIntersection(response.output.headers, result.headers, log);
+              // hapi wraps all error response in Boom object internally
+              response.output.headers = {
+                ...response.output.headers,
+                ...(result.headers as any), // hapi types don't specify string[] as valid value
+              };
+            } else {
+              findHeadersIntersection(response.headers, result.headers, log);
+              for (const [headerName, headerValue] of Object.entries(result.headers)) {
+                response.header(headerName, headerValue as any); // hapi types don't specify string[] as valid value
+              }
+            }
+          }
+        } else if (preResponseResult.isRender(result)) {
+          return responseToolkit.response(result.body).code(result.statusCode || statusCode);
```

Comment:
Let's remove `statusCode` from params If you don't use `statusCode` customization.

Reply:
Yeah I was wondering if I should remove it. I thought it was a nice feature to support, but you're right that this PR doesn't make use of it. But should I add `headers` then? If I remove `statusCode` it doesn't make sense to then add `headers` since the PR actually doesn't use any of them.

Reply:
I've removed support for setting a custom status code for now. If we want to support this we can always add it in a later PR.

Reply:
> But should I add headers then?

@watson I'm okay to postpone it as well